Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM golang:1.12-alpine as builder
+
+# Install SSL ca certificates
+RUN apk update && apk add git && apk add ca-certificates
+
+# The working directory should be /opt/test-runner
+WORKDIR /opt/test-runner
+RUN mkdir /opt/test-runner/output
+
+# Create appuser
+RUN adduser -D -g '' appuser
+RUN mkdir /home/appuser/.cache/go-build -p
+
+
+# source code
+COPY ./cmd/testoutput/main.go /opt/test-runner/main.go
+COPY ./go.mod /opt/test-runner/go.mod
+COPY ./go.sum /opt/test-runner/go.sum
+
+COPY ./exreport /opt/test-runner/exreport
+
+# download dependencies
+RUN go mod download
+
+# Create run.sh
+COPY ./run.sh /opt/test-runner/bin/run.sh
+RUN chmod +x /opt/test-runner/bin/run.sh
+
+# set file permissions
+RUN chown -R appuser /home/appuser/
+RUN chown -R appuser /opt/test-runner
+RUN chown -R appuser /home/appuser/
+RUN chown -R appuser /go/
+
+# build
+RUN go build /opt/test-runner/main.go
+RUN GOOS=linux GOARCH=amd64 go build --tags=build -o /go/bin/test-runner .
+
+# Build a minimal and secured container
+# The ast parser needs Go installed for import statements.
+# Therefore, unfortunately we cannot build from scratch as we would normally do with Go.
+FROM golang:1.12-alpine
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+COPY --from=builder /etc/passwd /etc/passwd
+COPY --from=builder /go/ /go/
```

Comment:
I'm not sure why you copy the entire go folder in. The point of this container is to keep it minimal and go is installed already. The only thing to copy from the `/go/bin` folder is the test-runner. If you build that to `/opt/test-runner` instead, you don't need any extra copy command for `/go/...`.